android/src/main/kotlin/com/amplitude/amplitude_flutter/AmplitudeFlutterPlugin.kt

```diff
@@ -263,236 +139,157 @@ class AmplitudeFlutterPlugin : FlutterPlugin, MethodCallHandler {
+        }
+        if (!jsonObject.getBoolean("city")) {
+            trackingOptions.disableCity()
+        }
+        if (!jsonObject.getBoolean("carrier")) {
+            trackingOptions.disableCarrier()
+        }
+        if (!jsonObject.getBoolean("deviceModel")) {
+            trackingOptions.disableDeviceModel()
+        }
+        if (!jsonObject.getBoolean("deviceManufacturer")) {
+            trackingOptions.disableDeviceManufacturer()
+        }
+        if (!jsonObject.getBoolean("osVersion")) {
+            trackingOptions.disableOsVersion()
+        }
+        if (!jsonObject.getBoolean("osName")) {
+            trackingOptions.disableOsName()
+        }
+        if (!jsonObject.getBoolean("adid")) {
+            trackingOptions.disableAdid()
+        }
+        if (!jsonObject.getBoolean("appSetId")) {
+            trackingOptions.disableAppSetId()
+        }
+        if (!jsonObject.getBoolean("deviceBrand")) {
+            trackingOptions.disableDeviceBrand()
+        }
+        if (!jsonObject.getBoolean("latLag")) {
+            trackingOptions.disableLatLng()
+        }
+        if (!jsonObject.getBoolean("apiLevel")) {
+            trackingOptions.disableApiLevel()
+        }
 
-                    // POSTINSERT
-                    "\$postInsert" -> {
-                        when (properties.get(key)) {
-                            is Int -> {
-                                identify.postInsert(key, properties.getInt(key))
-                            }
-                            is Long -> {
-                                identify.postInsert(key, properties.getLong(key))
-                            }
-                            is Double -> {
-                                identify.postInsert(key, properties.getDouble(key))
-                            }
-                            is String -> {
-                                identify.postInsert(key, properties.getString(key))
-                            }
-                            is Boolean -> {
-                                identify.postInsert(key, properties.getBoolean(key))
-                            }
-                            is JSONObject -> {
-                                identify.postInsert(key, properties.getJSONObject(key))
-                            }
-                            is JSONArray -> {
-                                identify.postInsert(key, properties.getJSONArray(key))
-                            }
-                        }
-                    }
+        return trackingOptions
+    }
 
-                    // UNSET
-                    "\$unset" -> {
-                        identify.unset(key)
-                    }
+    internal fun getEvent(json: JSONObject): BaseEvent {
+        val plan = json.getJSONObject("plan")
+        val ingestionMetadata = json.getJSONObject("ingestion_metadata")
+        val event = BaseEvent()
+        event.eventType = json.getString("event_type")
```

Comment:
Do we need to check if  the input has event_type or not?

Reply:
`eventType` is a required for `BaseEvent` constructor. So skip checking whether it's existing or not.